ambry-network/src/main/java/com/github/ambry/network/http2/Http2NetworkClient.java

```diff
@@ -86,18 +86,22 @@ public List<ResponseInfo> sendAndPoll(List<RequestInfo> requestsToSend, Set<Inte
       for (int correlationId : requestsToDrop) {
         Channel streamChannel = correlationIdInFlightToChannelMap.remove(correlationId);
         if (streamChannel != null) {
-          logger.warn("Drop request on streamChannel: {}", streamChannel);
+          RequestInfo requestInfo = streamChannel.attr(Http2NetworkClient.REQUEST_INFO).getAndSet(null);
           // Drop request just generates a ResponseInfo with TimeoutError to router.
           // The stream is still transmitting, but router will ignore ResponseInfo with the same correlationId.
           // We need stream reset to cancel the stream in transmitting.
-          RequestInfo requestInfo = streamChannel.attr(Http2NetworkClient.REQUEST_INFO).getAndSet(null);
           if (requestInfo != null) {
+            logger.warn("Drop request {}:{} on streamChannel: {}", requestInfo.getRequest().getRequestOrResponseType(),
+                correlationId, streamChannel);
             NetworkClientErrorCode errorCode = NetworkClientErrorCode.TimeoutError;
             if (http2ClientConfig.http2TimeoutAsNetworkError) {
               errorCode = NetworkClientErrorCode.NetworkError;
             }
             readyResponseInfos.add(new ResponseInfo(requestInfo, errorCode, null));
             ReferenceCountUtil.safeRelease(requestInfo.getRequest());
+            releaseAndCloseStreamChannel(streamChannel);
```

Comment:
The `releaseAndCloseStreamChannel` won't do anything here. Because we are already calling `streamChannel.attr(Http2NetworkClient.REQUEST_INFO).getAndSet(null);` in line 89 above. So within `releaseAndCloseStreamChannel`, the requestInfo will be null.

We should maybe call `releaseAndCloseStreamChannel` in line 89 itself?

Reply:
updated